### DISCUSSION TITLE
Use action based on commit instead of branch/tag

### DISCUSSION THREAD

**Discussion by javixeneize** [community/community#24957] [Actions]
<p>Hi</p>
<p>Is it possible to use an action referring to its commit instead of branch or tag?</p>
<p>A branch can have its content changed, and a tag can be deleted and re-released. This means that if i am using an action in my workflow, and the owner by mistake (or maliciously) changes the content of the branch or tag, then, my workflow will be impacted by that. However, if you refer to the action using a commit, this will be inmutable and it will not be affected by any of the changes mentioned above (other than deleting the commit, of course)</p>
<p>Thanks</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can do that, example <a href="https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#using-shas">from the docs</a>:</p>
<pre><code class="lang-yaml">steps:
    - uses: actions/javascript-action@172239021f7ba04fe7327647b213799853a9eb89
</code></pre>
